frontend/src/libs/components/dropdown-multi-select/dropdown-multi-select.tsx

```diff
@@ -0,0 +1,138 @@
+import {
+  type Control,
+  type ControllerRenderProps,
+  type FieldErrors,
+  type FieldPath,
+  type FieldValues,
+} from 'react-hook-form';
+import Select, {
+  type ClassNamesConfig,
+  type GroupBase,
+  type MultiValue,
+  type SingleValue,
+} from 'react-select';
+
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import {
+  useCallback,
+  useFormController,
+  useMemo,
+  useState,
+} from '~/libs/hooks/hooks.js';
+import { type SelectOption } from '~/libs/types/select-option.type.js';
+
+import styles from './styles.module.scss';
+
+type Properties<T extends FieldValues> = {
+  options: SelectOption[];
+  name?: FieldPath<T>;
+  control?: Control<T, null>;
+  errors: FieldErrors<T>;
+  label?: string;
+  onChange?: (option: MultiValue<T>) => void;
+  placeholder?: string;
+  field?: ControllerRenderProps<T, FieldPath<T>>;
+  className?: string;
+};
+
+const getClassNames = (
+  isMenuOpen: boolean,
+): ClassNamesConfig<SelectOption, true, GroupBase<SelectOption>> => ({
+  container: () => styles.container,
+  control: () => getValidClassNames(styles.control, styles.multiControl),
+  multiValueLabel: () => styles.multiValueLabel,
+  option: () => styles.option,
+  menu: () => styles.singleValue,
+  placeholder: () => styles.placeholder,
+  singleValue: () => styles.singleValue,
+  valueContainer: () => styles.valueContainer,
+  dropdownIndicator: () =>
+    isMenuOpen
+      ? getValidClassNames(styles.dropdownIndicator, styles.upside)
+      : styles.dropdownIndicator,
+  indicatorSeparator: () => styles.indicatorSeparator,
+});
+
+const DropdownMultiSelect = <T extends FieldValues>({
```

Comment:
we have a dropdown component, can you extend it to multi dropdown ? or why do you created almost a copy of it